test/ProductConstructionService.DependencyFlow.Tests/TestsWithServices.cs

```diff
@@ -27,7 +27,10 @@ protected async Task Execute(Func<IServiceProvider, Task> run)
         using (ServiceProvider container = services.BuildServiceProvider())
         using (IServiceScope scope = container.GetRequiredService<IServiceScopeFactory>().CreateScope())
         {
-            await BeforeExecute(scope.ServiceProvider);
+            if (initialize)
```

Comment:
I think the intended use was that we do not exercise the state machine several times in a row but only once.
That means that we do not call `WhenUpdateIsProcessed` three times but rather have three tests moving from the various intermediate states to the next.
I think the same issue appeared in the `GivenANewBuild`.

But I guess we can leave it like this now, since it's done.

Reply:
Thinking about it now, I don't think we gain anything by calling the method twice, since the state doesn't change between